phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java

```diff
@@ -2846,21 +2872,25 @@ private MetaDataMutationResult mutateColumn(
                 separateLocalAndRemoteMutations(region, tableMetadata, localMutations,
                         remoteMutations);
                 if (!remoteMutations.isEmpty()) {
-                    // there should only be remote mutations if we are adding a column to a view
+                    // there should only be remote mutations if we are updating the last ddl
+                    // timestamp for child views, or we are adding a column to a view
                     // that uses encoded column qualifiers (the remote mutations are to update the
                     // encoded column qualifier counter on the parent table)
-                    if (mutator.getMutateColumnType() == ColumnMutator.MutateColumnType.ADD_COLUMN
+                    if (childViews.size() > 0 || ( mutator.getMutateColumnType() == ColumnMutator.MutateColumnType.ADD_COLUMN
```

Comment:
Shouldn't this be done even if `if (!remoteMutations.isEmpty())` is false if the table/view has child views??

Reply:
I added the childViews.size() check when I was sending the ddl timestamp mutations remotely to the child view header rows, which I stopped doing in the last draft. That part of the if clause should just be removed right? Or was the existing logic wrong if a child view existed?